django_tenants/middleware/main.py

```diff
@@ -54,7 +55,17 @@ def process_request(self, request):
     def no_tenant_found(self, request, hostname):
         """ What should happen if no tenant is found.
         This makes it easier if you want to override the default behavior """
-        if hasattr(settings, 'SHOW_PUBLIC_IF_NO_TENANT_FOUND') and settings.SHOW_PUBLIC_IF_NO_TENANT_FOUND:
+        if hasattr(settings, 'DEFAULT_NOT_FOUND_TENANT_VIEW'):
+            view_path = settings.DEFAULT_NOT_FOUND_TENANT_VIEW
+            view = import_string(view_path)
+            if hasattr(view, 'as_view'):
+                response = view.as_view()(request)
+            else:
+                response = view(request)
+            if hasattr(response, 'render'):
+                response.render()
+            return response
+        elif hasattr(settings, 'SHOW_PUBLIC_IF_NO_TENANT_FOUND') and settings.SHOW_PUBLIC_IF_NO_TENANT_FOUND:
```

Comment:
Can you exercise this setting as part of the test suite ?

Reply:
🫡 done